certificate/authorization/db_test.go

```diff
@@ -114,25 +103,58 @@ func TestAuthorizationDB_Create(t *testing.T) {
 			}
 
 			expectedAuths, err := authDB.Create(ctx, testCase.email, testCase.incCount)
-			if testCase.errClass != nil {
-				assert.True(t, testCase.errClass.Has(err))
-			}
-			if testCase.err != nil {
-				assert.Equal(t, testCase.err, err)
-			}
-			if testCase.errClass == nil && testCase.err == nil {
-				assert.NoError(t, err)
-			}
-			assert.Len(t, expectedAuths, testCase.newCount)
+			require.NoError(t, err)
+			require.Len(t, expectedAuths, testCase.newCount)
 
 			v, err := authDB.db.Get(ctx, emailKey)
-			assert.NoError(t, err)
-			assert.NotEmpty(t, v)
+			require.NoError(t, err)
+			require.NotEmpty(t, v)
 
 			var actualAuths Group
 			err = actualAuths.Unmarshal(v)
-			assert.NoError(t, err)
-			assert.Len(t, actualAuths, testCase.endCount)
+			require.NoError(t, err)
+			require.Len(t, actualAuths, testCase.endCount)
+		})
+	}
+}
+
+func TestAuthorizationDB_Create_error(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	authDB := newTestAuthDB(t, ctx)
+	defer ctx.Check(authDB.Close)
+
+	cases := []struct {
+		testID,
+		email string
+		count    int
+		errClass *errs.Class
+		err      error
+	}{
+		{
+			"empty userID",
+			"", 1,
+			&ErrDB,
+			ErrEmptyUserID,
+		},
+		{
+			"negative count",
+			"user@mail.example", -1,
+			&ErrDB,
+			ErrCount,
+		},
+	}
+
+	for _, c := range cases {
+		testCase := c
+		t.Run(c.testID, func(t *testing.T) {
+			auths, err := authDB.Create(ctx, testCase.email, testCase.count)
+			if !assert.True(t, testCase.errClass.Has(err)) {
```

Comment:
You can simplify this with

`assert.Truef(t, testCase.errClass.Has(err), "error: %s", err)`